_pages/algorithm-development/languages/python.md

```diff
@@ -10,57 +10,90 @@ image:
 
-Also, algorithms can call other algorithms and manage data on the Algorithmia platform. You can learn more about calling algorithms in the <a href="{{site.baseurl}}/clients/python">Algorithmia Python Client Guide</a>.
+Optionally, an algorithm can also have a `load()` function, where you can prepare your algorithm for runtime operations, such as model loading, configuration, etc. 
 
-## Write your First Algorithm
+Algorithms must also contain a call to the handler function with your `apply()` and optional `load()` function as inputs. This will convert the project into an executable, rather than a library, which interacts with the `langserver` service on Algorithmia while also being debuggable via `stdin`/`stdout` when executed outside of the Algorithmia container. An `init()` function starts the algorithm and allows you to provide an input for use when the algorithm is executed locally, bypassing `stdin` parsing and simplifying debugging. You can also step through your algorithm in your IDE of choice by executing your `src/Algorithm.py` script.
 
-If you've followed the <a href="{{site.baseurl}}/algorithm-development/algorithm-basics/your-first-algo">Getting Started Guide</a>, you'll notice in your algorithm editor, there is boilerplate code that returns "Hello" and whatever you input to the console.
+Let's look at an example to clarify some of these concepts.
 
-The main thing to note about the algorithm is that it's wrapped in the apply() function.
+## Hello World
 
-The apply() function defines the input point of the algorithm. We use the apply() function in order to make different algorithms standardized. This makes them easily chained and helps authors think about designing their algorithms in a way that makes them easy to leverage and predictable for end users.
-{: .notice-info}
+Below you'll find a `src/Algorithm.py` file which prints "Hello" plus an input when it is invoked. We start by importing the Algorithmia ADK, and then defining our `apply` function, followed by our call to the handler function `ADK()`, and finally calling `init()` to start the function.
+
+{% highlight python %}
+from Algorithmia import ADK
+
+def apply(input):
+    return "hello {}".format(str(input))
+
+algorithm = ADK(apply)
+algorithm.init("Algorithmia")
+{% endhighlight %}
 
-Go ahead and remove the boilerplate code below that's inside the apply() function because we'll be writing a different algorithm in this tutorial:
+When executed in an Algorithmia container and providing the string "HAL 9000" as an input, this algorithm will output "Hello HAL 9000". If executed locally, the algorithm will print `Hello Algorithmia` to `stdout` instead.
```

Comment:
change "in an Algorithmia container" --> "on the Algorithmia platform" for clarity

Reply:
also change instances of `Hello` --> `hello` according to source code